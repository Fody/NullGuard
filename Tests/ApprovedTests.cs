#if (NET472)
using VerifyXunit;
using Fody;
using Xunit;
using System.Linq;
using System.Threading.Tasks;

[UsesVerify]
public class ApprovedTests
{
    [Fact]
    public Task ClassWithBadAttributes()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithBadAttributes"));
    }

    [Fact]
    public Task ClassWithPrivateMethod()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithPrivateMethod"));
    }

    [Fact]
    public Task ClassWithPrivateMethodNoAssert()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithPrivateMethod"));
    }

    [Fact]
    public Task GenericClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClass`1"));
    }

    [Fact]
    public Task GenericClassWithValueTypeConstraint()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClassWithValueTypeConstraints`1"));
    }

    [Fact]
    public Task GenericClassWithReferenceTypeConstraints()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClassWithReferenceTypeConstraints`1"));
    }

    [Fact]
    public Task Indexers()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "Indexers"));
    }

    [Fact]
    public Task InterfaceBadAttributes()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "InterfaceBadAttributes"));
    }

    [Fact]
    public Task SimpleClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleClass"));
    }

    [Fact]
    public Task SimpleClassNoAssert()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleClass"));
    }

    [Fact]
    public Task SkipIXamlMetadataProvider()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "XamlMetadataProvider"));
    }

#if (DEBUG)
    [Fact]
    public Task SpecialClass_debug()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SpecialClass"));
    }
#else
    [Fact]
    public Task SpecialClass_release()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SpecialClass"));
    }
#endif

    [Fact]
    public Task PublicNestedInsideNonPublic()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "NonPublicWithNested"));
    }

    [Fact]
    public Task UnsafeClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "UnsafeClass"));
    }

    [Fact]
    public Task ClassWithImplicitInterface()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithImplicitInterface"));
    }

    [Fact]
    public Task ClassWithExplicitInterface()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithExplicitInterface"));
    }

    [Fact]
    public Task InfosList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Messages.Select(x=>x.Text));
    }

    [Fact]
    public Task WarnsList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Warnings.Select(x=>x.Text));
    }

    [Fact]
    public Task ErrorsList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Errors.Select(x=>x.Text));
    }
}

#endif