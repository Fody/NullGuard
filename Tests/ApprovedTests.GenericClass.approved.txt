.class public auto ansi beforefieldinit GenericClass`1<T>
       extends [mscorlib]System.Object
{
  .field private !T '<NonNullProperty>k__BackingField'
  .method public hidebysig specialname instance !T 
          get_NonNullProperty() cil managed
  {
    // Code size       27 (0x1b)
    .maxstack  2
    IL_0000:  ldarg.0
    IL_0001:  ldfld      !0 class GenericClass`1<!T>::'<NonNullProperty>k__BackingField'
    IL_0006:  dup
    IL_0007:  box        !T
    IL_000c:  brtrue.s   IL_001a
    IL_000e:  pop
    IL_000f:  ldstr      "[NullGuard] Return value of property 'T GenericCla"
    + "ss`1::NonNullProperty()' is null."
    IL_0014:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
    IL_0019:  throw
    IL_001a:  ret
  } 
  .method public hidebysig specialname instance void 
          set_NonNullProperty(!T 'value') cil managed
  {
    // Code size       32 (0x20)
    .maxstack  2
    IL_0000:  ldarg.1
    IL_0001:  box        !T
    IL_0006:  brtrue.s   IL_0018
    IL_0008:  ldstr      "value"
    IL_000d:  ldstr      "[NullGuard] Cannot set the value of property 'T Ge"
    + "nericClass`1::NonNullProperty()' to null."
    IL_0012:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_0017:  throw
    IL_0018:  ldarg.0
    IL_0019:  ldarg.1
    IL_001a:  stfld      !0 class GenericClass`1<!T>::'<NonNullProperty>k__BackingField'
    IL_001f:  ret
  } 
  .method public hidebysig instance !T  NonNullMethod() cil managed
  {
    // Code size       30 (0x1e)
    .maxstack  2
    .locals init ([0] !T __var_0)
    IL_0000:  ldloca.s   __var_0
    IL_0002:  initobj    !T
    IL_0008:  ldloc.0
    IL_0009:  dup
    IL_000a:  box        !T
    IL_000f:  brtrue.s   IL_001d
    IL_0011:  pop
    IL_0012:  ldstr      "[NullGuard] Return value of method 'T GenericClass"
    + "`1::NonNullMethod()' is null."
    IL_0017:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
    IL_001c:  throw
    IL_001d:  ret
  } 
  .method public hidebysig instance !!U  GenericMethod<U>(!T t,
                                                          !!U u) cil managed
  {
    // Code size       78 (0x4e)
    .maxstack  2
    .locals init ([0] !!U __var_0)
    IL_0000:  ldarg.1
    IL_0001:  box        !T
    IL_0006:  brtrue.s   IL_0018
    IL_0008:  ldstr      "t"
    IL_000d:  ldstr      "[NullGuard] t is null."
    IL_0012:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_0017:  throw
    IL_0018:  ldarg.2
    IL_0019:  box        !!U
    IL_001e:  brtrue.s   IL_0030
    IL_0020:  ldstr      "u"
    IL_0025:  ldstr      "[NullGuard] u is null."
    IL_002a:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
                                                                                     string)
    IL_002f:  throw
    IL_0030:  ldloca.s   __var_0
    IL_0032:  initobj    !!U
    IL_0038:  ldloc.0
    IL_0039:  dup
    IL_003a:  box        !!U
    IL_003f:  brtrue.s   IL_004d
    IL_0041:  pop
    IL_0042:  ldstr      "[NullGuard] Return value of method 'U GenericClass"
    + "`1::GenericMethod(T,U)' is null."
    IL_0047:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
    IL_004c:  throw
    IL_004d:  ret
  } 
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       7 (0x7)
    .maxstack  1
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  ret
  } 
  .property instance !T NonNullProperty()
  {
  } 
}
